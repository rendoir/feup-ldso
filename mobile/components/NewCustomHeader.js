import React from 'react';
import {
    StatusBar,
    StyleSheet
} from 'react-native';
import {
    View,
    Text,
    Icon
} from 'native-base';
import SwitchToggle from 'react-native-switch-toggle';
import '../global.js';

export default class NewCustomHeader extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {

        return (
            <View style={styles.wrapperView}>
                <StatusBar hidden />
                <View style={styles.mainView}>
                    <View style={styles.goBackIconView}>
                        <Icon
                            className='goBack'
                            onPress={ () => this.props.navigation.goBack(null) }
                            name='arrow-circle-left'
                            type='FontAwesome'
                            style={styles.goBackIcon} />
                    </View>
                    <View style={styles.textView} >
                        <Text style={styles.text}>
                            {this.props.text}
                        </Text>
                    </View>
                    <View style={styles.toggleView}>
                        <SwitchToggle
                            buttonText={this.props.language === 'EN' ? 'EN' : ''}
                            backTextRight={"PT"}
                            backTextLeft={"EN"}
                            switchOn={this.props.language === 'PT'}
                            onPress={this.props.toggleLanguage}
                            type={1}
                            buttonStyle={{
                                alignItems: 'center',
                                justifyContent: 'center',
                                position: 'absolute'
                            }}
                            rightContainerStyle={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}
                            leftContainerStyle={{ flex: 1, alignItems: 'center', justifyContent: 'flex-start' }}

                            buttonTextStyle={{ fontSize: 9 }}
                            textRightStyle={{ fontSize: 9 }}
                            textLeftStyle={{ fontSize: 9 }}
                            backgroundColorOn='#fff'
                            backgroundColorOff='#fff'
                            circleColorOff='#e5e1e0'
                            circleColorOn='#e5e1e0'
                            className="change-language"
                            accessibilityLabel="Toggle language"

                        />
                    </View>
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    wrapperView: {
        height: '6%'
    },
    mainView: {
        backgroundColor: '#455A64',
        height: '100%',
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    goBackIconView: {
        flex: 1,
        justifyContent: 'center',
        marginHorizontal: '2%',
        alignItems: 'flex-start',
        alignSelf: 'center'
    },
    goBackIcon: {
        color: '#D05722',
        fontSize: 28
    },
    toggleView: {
        flex: 1,
        alignItems: 'flex-end',
        marginRight: '2%',
        alignSelf: 'center'
    },
    textView: {
        flex: 1,
        alignSelf: 'center',
        alignItems: 'center'
    },
    text: {
        color: '#F0F0F0',
        textAlign: 'center'
    }
});
