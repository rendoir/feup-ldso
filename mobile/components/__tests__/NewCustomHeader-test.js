/**
 * @jest-environment jsdom
 */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import NewCustomHeader from '../NewCustomHeader';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


describe('Custom Header', () => {

    it('renders correctly in portuguese', () => {
        const tree = renderer.create(<NewCustomHeader language='PT' />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly in english', () => {
        const tree = renderer.create(<NewCustomHeader language='EN' />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('change language', () => {
        let toggleLanguage = jest.fn();
        const wrapper = shallow(<NewCustomHeader language='PT' toggleLanguage={toggleLanguage}/>);

        let button = wrapper.find('.change-language').first();
        button.props().onPress();

        expect(toggleLanguage).toHaveBeenCalled();
    });

    it('test go back', () => {
        const navigation = { getParam: jest.fn(), goBack: jest.fn() };
        const wrapper = shallow(<NewCustomHeader language='PT' navigation={navigation} />);

        let button = wrapper.find('.goBack').first();
        button.props().onPress();

        expect(navigation.goBack).toHaveBeenCalled();
    });

});
