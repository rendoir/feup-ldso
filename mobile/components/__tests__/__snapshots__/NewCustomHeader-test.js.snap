// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Header renders correctly in english 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "height": "6%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#455A64",
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginHorizontal": "2%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#F0F0F0",
              "textAlign": "center",
            },
          ]
        }
        uppercase={false}
      />
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "marginRight": "2%",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "borderRadius": 18,
              "flexDirection": "row",
              "height": 36,
              "padding": 3,
              "width": 72,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 9,
                }
              }
            >
              EN
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(229, 225, 224, 1)",
                "borderRadius": 15,
                "height": 30,
                "justifyContent": "center",
                "position": "absolute",
                "transform": Array [
                  Object {
                    "translateX": 6,
                  },
                ],
                "width": 30,
              }
            }
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 9,
                  }
                }
              >
                EN
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 9,
                }
              }
            >
              PT
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Custom Header renders correctly in portuguese 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "height": "6%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#455A64",
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginHorizontal": "2%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#F0F0F0",
              "textAlign": "center",
            },
          ]
        }
        uppercase={false}
      />
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "marginRight": "2%",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "borderRadius": 18,
              "flexDirection": "row",
              "height": 36,
              "padding": 3,
              "width": 72,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 9,
                }
              }
            >
              EN
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(229, 225, 224, 1)",
                "borderRadius": 15,
                "height": 30,
                "justifyContent": "center",
                "position": "absolute",
                "transform": Array [
                  Object {
                    "translateX": 36,
                  },
                ],
                "width": 30,
              }
            }
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 9,
                  }
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 9,
                }
              }
            >
              PT
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
